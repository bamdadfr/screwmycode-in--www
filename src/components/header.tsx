import React, { ReactElement } from 'react'
import { sendToast } from './toast.utils'
import './header.styles.css'
import { IconRepeat, IconToggle, IconQuestion } from './icons'
import { stateMap } from './state.map'

const HeaderToggle = stateMap ((props: any): ReactElement => {

    const { three, setStateThreeVisible } = props
    const onClick = (e: boolean): void => setStateThreeVisible (e)

    return (
        <>
            <span
                onClick={(): void => onClick (!three.visible)}
                onKeyDown={(): void => undefined}
                role="button"
                tabIndex={-1}
            >
                {three.visible ? <IconToggle.on /> : <IconToggle.off />}
            </span>
        </>
    )

})

const HeaderFaq = (): ReactElement => {

    return (
        <>
            <span
                onClick={(): void => sendToast ('FAQ coming soon')}
                onKeyDown={(): void => undefined}
                role="button"
                tabIndex={-1}
            >
                <IconQuestion />
            </span>

        </>
    )

}

const HeaderRepeat = (): ReactElement => {

    return (
        <>
            <span
                onClick={(): void => sendToast ('repeat mode coming soon')}
                onKeyDown={(): void => undefined}
                role="button"
                tabIndex={-1}
            >
                <IconRepeat />
            </span>
        </>

    )

}

export const Header = (): ReactElement => {

    return (
        <div className="corner-help">
            <HeaderFaq />
            <HeaderToggle />
            <HeaderRepeat />
        </div>
    )

}