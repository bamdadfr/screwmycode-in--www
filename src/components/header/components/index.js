export { IconHome } from './icon-home'

export { IconRepeatOff } from './icon-repeat-off'

export { IconRepeatOn } from './icon-repeat-on'