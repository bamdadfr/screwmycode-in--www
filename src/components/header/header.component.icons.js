import React from 'react'
import { Icon } from '@iconify/react'
import repeatOff from '@iconify/icons-mdi/repeat-off'
import repeat from '@iconify/icons-mdi/repeat'
import home from '@iconify/icons-mdi/home'

export const IconRepeat = {
    'On': () => <Icon icon={repeat}/>,
    'Off': () => <Icon icon={repeatOff}/>,
}

export const IconHome = () => <Icon icon={home}/>