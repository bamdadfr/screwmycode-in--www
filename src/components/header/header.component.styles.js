import styled from 'styled-components'

export const Container = styled.div`
    position: fixed;
    z-index: 10;
    margin: 7px 7px;
    top: 0;
    left: 0;
    font-size: 2rem;
    color: ${(props) => props.theme.primary};

    > button {
        transition: color 200ms ease-in-out;

        &:hover {
            cursor: pointer;
            color: ${(props) => props.theme.highlight};
        }

    }
`

