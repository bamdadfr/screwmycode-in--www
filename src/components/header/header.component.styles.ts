import styled from 'styled-components';
import {Icon} from '@iconify/react';
import {flexCenter, hoverLink} from 'src/app/styles/styles';

const paddingHorizontal = 7;
const gap = 3;

export const Container = styled.div`
  ${flexCenter};
  gap: ${gap}px;

  padding: ${gap}px ${paddingHorizontal}px;

  background: ${(props) => props.theme.background.primary};
  color: ${(props) => props.theme.primary};

  font-size: 2em;
`;

export const Side = styled.div`
  ${flexCenter};
  gap: ${gap}px;

  position: fixed;
  ${hoverLink};
`;

export const Left = styled(Side)`
  left: ${paddingHorizontal}px;
`;

export const Right = styled(Side)`
  right: ${paddingHorizontal}px;
`;

export const Podium = styled(Icon)`
  font-size: 91%;
  transform: translate(4px, -3px);
`;
