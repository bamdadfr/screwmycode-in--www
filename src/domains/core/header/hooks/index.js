export { useHeaderComponent } from './use-header-component'