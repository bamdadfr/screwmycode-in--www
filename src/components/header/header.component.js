import React from 'react'
import Link from 'next/link'
import { IconHome, IconRepeatOff, IconRepeatOn } from './components'
import { Container } from './header.component.styles'
import { useHeaderComponent } from './hooks'

/**
 * @returns {React.ReactElement} react component
 */
export function HeaderComponent () {

    const { isRepeat, toggleRepeat } = useHeaderComponent ()

    return (
        <>
            <Container>
                <Link href="/">
                    <button
                        type="button"
                        aria-label="home"
                    >
                        <IconHome/>
                    </button>
                </Link>
                <button
                    type="button"
                    aria-label="repeat"
                    onClick={() => toggleRepeat ()}
                    value={isRepeat ? 'on' : 'off'}
                >
                    {isRepeat
                        ? <IconRepeatOn/>
                        : <IconRepeatOff/>
                    }
                </button>
            </Container>
        </>
    )

}