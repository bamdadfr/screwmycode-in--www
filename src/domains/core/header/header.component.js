import React from 'react'
import Link from 'next/link'
import { IconRepeat, IconHome } from './header.icons'
import { Container } from './header.styles'
import { useHeaderComponent } from './hooks'

/**
 * @function
 * @name HeaderComponent
 * @description layout menu component
 * @returns {React.ReactElement} - react component
 */
export function HeaderComponent () {

    const { repeat, toggleRepeat } = useHeaderComponent ()

    return (
        <>
            <Container>
                <Link href="/">
                    <button
                        type="button"
                        aria-label="home"
                    >
                        <IconHome/>
                    </button>
                </Link>
                <button
                    type="button"
                    aria-label="repeat"
                    onClick={() => toggleRepeat ()}
                >
                    {repeat ? <IconRepeat.On/> : <IconRepeat.Off/>}
                </button>
            </Container>
        </>
    )

}