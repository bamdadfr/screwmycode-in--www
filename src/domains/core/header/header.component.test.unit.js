import React from 'react'
import { HeaderComponent } from './header.component'
import { JestInstance } from '../../../../jest/jest-instance'

const instance = JestInstance (<HeaderComponent/>)

it ('should mount and match', () => {

    expect (instance).toBeDefined ()

    expect (instance).toMatchSnapshot ()

})

it ('should have a classname prop from styled components', () => {

    expect (instance.props.className).toBeDefined ()

    expect (instance.props.className).toContain ('sc')

})

it ('should have 2 children', () => {

    expect (instance.children).toHaveLength (2)

})
