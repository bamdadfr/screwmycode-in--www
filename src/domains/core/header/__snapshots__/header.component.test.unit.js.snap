// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should mount and match 1`] = `
<div
  className="sc-1w34k6z-0 iubAlC"
>
  <button
    aria-label="home"
    onClick={[Function]}
    onMouseEnter={[Function]}
    type="button"
  >
    <svg
      dangerouslySetInnerHTML={
        Object {
          "__html": "<path d=\\"M10 20v-6h4v6h5v-8h3L12 3L2 12h3v8h5z\\" fill=\\"currentColor\\"/>",
        }
      }
      focusable={false}
      height="1em"
      preserveAspectRatio="xMidYMid meet"
      style={
        Object {
          "transform": "rotate(360deg)",
        }
      }
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    />
  </button>
  <button
    aria-label="repeat"
    onClick={[Function]}
    type="button"
  >
    <svg
      dangerouslySetInnerHTML={
        Object {
          "__html": "<path d=\\"M17 17H7v-3l-4 4l4 4v-3h12v-6h-2M7 7h10v3l4-4l-4-4v3H5v6h2V7z\\" fill=\\"currentColor\\"/>",
        }
      }
      focusable={false}
      height="1em"
      preserveAspectRatio="xMidYMid meet"
      style={
        Object {
          "transform": "rotate(360deg)",
        }
      }
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    />
  </button>
</div>
`;
